### PR TITLE
refactor: move off `@apidevtools/openapi-schema`

### PR DESCRIPTION
**Pull request by darrenyong**
| 🚥 Resolves RM-9343 |
| :------------------- |

## 🧰 Changes
The schemas we were validating against hasn't been updated in ages and there's a bug with a specific version where it's validating against `descriptions` instead of `description`. This makes it so that adding a valid `description` is considered invalid. 

This is being changed to use our [fork](https://github.com/readmeio/openapi-schemas) of `@apidevtools/openapi-schema`.

## 🧬 QA & Testing

Provide as much information as you can on how to test what you've done.

### CODE REVIEW

**Comment by erunion** [comment]
Not that I want us to have to fork and maintain more open source repos but it might be cleaner for us to fork `@apidevtools/openapi-schema` to keep it up to date? I'm a bit hesitant to wholesale pull these schemas over here without an easy way to update them, which `@apidevtools/openapi-schema` has support for with some `npm run-script` tooling.

**Review by erunion — APPROVED** [review]
cool if you just want to do this tho, let me know and i can set up a fork or if you want me to merge + publish this

**Comment by darrenyong** [comment]
opting to fork the repo. closing this out!

**Comment by darrenyong** [comment]
- [ ] this [spec](https://github.com/readmeio/openapi-parser/blob/ff9523c0867158186301825d58931802c5b14808/test/fixtures/real-world-apis.json#L88-L90) ([link to OAS](https://api.apis.guru/v2/specs/adyen.com/CheckoutService/70/openapi.yaml)) is failing off [this specific](https://github.com/OAI/OpenAPI-Specification/blob/209d9a4407097448da4e63f70dd60d66936a81d8/schemas/v3.1/schema.json#L967-L969) jsonschema validation block. need to look into it

File: package.json [hunk 1]
```diff
@@ -66,6 +66,7 @@
     "@jsdevtools/ono": "^7.1.3",
     "@readme/better-ajv-errors": "^1.6.0",
     "@readme/json-schema-ref-parser": "^1.2.0",
+    "@readme/openapi-schemas": "^3.0.1",
```
Comment (erunion) [inline package.json:69]: can you install `@apidevtools/openapi-schemas`?

**Review by erunion — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (darrenyong) [reply package.json:68]: uninstalled!

**Comment by darrenyong** [comment]
https://github.com/readmeio/openapi-schemas/pull/2

**Comment by darrenyong** [comment]
tagged you guys @Dashron and @julshotal since Jon and Kanad are OOO for the week!

File: package.json [hunk 2]
```diff
@@ -61,11 +61,11 @@
     "typescript": "^5.4.5"
   },
   "dependencies": {
-    "@apidevtools/openapi-schemas": "^2.1.0",
     "@apidevtools/swagger-methods": "^3.0.2",
     "@jsdevtools/ono": "^7.1.3",
     "@readme/better-ajv-errors": "^1.6.0",
     "@readme/json-schema-ref-parser": "^1.2.0",
+    "@readme/openapi-schemas": "^3.0.1",
```
Comment (darrenyong) [inline package.json:68]: if the newest version of `@readme/openapi-schemas` is `3.1.0`, does the dependency version also need to be updated?

**Comment by darrenyong** [comment]
@erunion, some of these tests take a while to run since there are some pretty large OAS files so they're timing out. is this something we need to change via `--testTimeout`?

File: package.json [hunk 3]
```diff
@@ -43,7 +43,7 @@
     "pretest": "npm run lint",
     "prettier": "prettier --check .",
     "prettier:write": "prettier --check --write .",
-    "test": "vitest run --coverage"
+    "test": "vitest run --coverage --testTimeout=9000"
```
Comment (erunion) [inline package.json:46]: should be able to remove this timeout if you rebase the latest `main`

File: package.json (hunk 2 above)
Comment (erunion) [reply package.json:65]: not sure i follow

**Review by erunion — APPROVED** [review]

**Pull request merged**